rfcs/20230818-Dropout/README.md

```diff
@@ -0,0 +1,68 @@
+# Introducing Dropout primitive attribute
+
+## Introduction
+
+In many DNN and GNN models [Dropout](https://en.wikipedia.org/wiki/Convolutional_neural_network#Dropout) is used to improve training results, in some cases this layer can take big ammount of time. To improve performance of training of such models, we want to optimize it and in result fuse with previous primitive.
+
+This idea was already proposed some time ago, as a result of that discussion, primitive attribute is proposed to resolve such performance gap.
+
+## Proposal
+
+Additional function to set dropout attibute in C:
+
+```c
+/// Returns probability for drop-out primitive attribute.
+///
+/// @param attr Primitive attributes.
+/// @param p Drop-out probability
+/// @returns #dnnl_success on success and a status describing the error
+///     otherwise.
+dnnl_status_t DNNL_API dnnl_primitive_attr_get_dropout(
+        const_dnnl_primitive_attr_t attr,
+        float *p, const_dnnl_memory_desc_t *drop_desc);
```

Comment:
Yes, mentioned algorithm is what we expect from this primitive attiribute - compute and apply mask.
On backward pass mask from forward will be multiplied by input(see [native_dropout_backward from PyTorch](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Dropout.cpp#L130))
The main idea was to move mask generation inside of oneDNN. Otherwise user can just use binary_mul post op to apply existing mask.

Reply:
Thanks. That clarifies the scope of the rfc (the new attribute is for fwd only).

Reply:
BTW, what datatypes is oneDNN expected to support for the mask?